connector/elasticapmconnector/config.go

```diff
@@ -66,6 +71,26 @@ type AggregationConfig struct {
 	Intervals []time.Duration `mapstructure:"intervals"`
 }
 
+// CustomConfig defines customizations for the metrics produced by the connector.
+type CustomConfig struct {
+	// ResourceAttributes define a list of resource attributes that will be added
+	// to the aggregated metrics as optional attributes i.e. the attribute will be
+	// added to the aggregated metrics if they are present in the incoming signal,
+	// otherwise, the attribute will be ignored.
+	//
+	// NOTE: any custom attributes should have a bounded and preferably low
+	// cardinality to be performant.
+	ResourceAttributes []string `mapstructure:"custom_resource_attributes"`
+	// Attributes define a list of resource attributes that will be added to the
+	// aggregated metrics as optional attributes i.e. the attribute will be added
+	// to the aggregated metrics if they are present in the incoming signal,
+	// otherwise, the attribute will be ignored.
+	//
+	// NOTE: any custom attributes should have a bounded and preferably low
+	// cardinality to be performant.
+	Attributes []string `mapstructure:"custom_attributes"`
```

Comment:
[**For reviewers**] (CC: @AlexanderWert @axw ) I have tried to keep the customization for the attributes minimal - the configuration only takes a list of attributes and considers them optional i.e. if the attribute is present in the incoming signal then it is added to the aggregated metric, otherwise it is ignored. Features like default values (which is present in signal to metrics component) are deliberately excluded. Let me know your thoughts on this. 

Note that adding the advanced features is a simple task, so, if we feel like we should have them I will make the changes.

Reply:
In addition, if we want to keep the configurations even simpler i.e. just take a list of resource attributes and other attributes and add them to all metrics then that can be done too.

I am not sure how much flexibility we should give to the end user as it might have a negative impact on us in terms of maintenance as well as users breaking the expectations.

Reply:
That sounds good to me. @AlexanderWert WDYT? Does the simplified model meet the user's requirement?

Reply:
Rethinking, service_summary can be customized using the resource attributes and span, dp, log record attributes doesn't make much sense for service summary. I am going ahead with the suggested simplified implementation for now to unblock this issue.